include/smack/SmackRep.h

```diff
@@ -96,6 +97,9 @@ class SmackRep {
 
   bool isUnsafeFloatAccess(const llvm::Type* elemTy, const llvm::Type* resultTy);
 
+  static std::string indexedName(std::string name, std::initializer_list<std::string> idxs);
```

Comment:
It is not clear why any of these are member functions, since they do not access members of the SmackRep class. Shouldn’t they just be regular (non-member) functions?

Reply:
These functions are used by both SmackRep.cpp and Prelude.cpp. I can duplicate them in both files, but this requires placing the ones in Prelude.cpp outside of the smack namespace. This is what I've done in [this commit](https://github.com/smackers/smack/pull/410/commits/004fd28b2afab765d6c2307f35277a33e8922ea7).

Reply:
Why does this require duplication, or moving outside of the smack namespace? Why can’t SmackRep.h simply declare these functions in the smack namespace, let SmackRep.cpp define them, and then have Prelude.cpp refer to the functions declared in SmackRep.h?

Maybe I am missing something subtle about namespaces?

Reply:
No, it was my mistake. I just made [another commit](https://github.com/smackers/smack/pull/410/commits/0399e46b539414fe7778f67adfafd52c0a52b4d0) that implements your suggestion.

Reply:
Nice! I’ll add just a couple more comments.